registry/registry.go

```diff
@@ -38,24 +40,24 @@ type Registry struct {
  * This passes http.DefaultTransport to WrapTransport when creating the
  * http.Client.
  */
-func New(registryUrl, username, password string) (*Registry, error) {
+func New(ctx context.Context, registryUrl, username, password string) (*Registry, error) {
```

Comment:
It's what implements

```

resource "dockerregistry_image" "apprunner" {
  repository = "mdg0/apprunner"
  tag = "20160606T225805Z_86c9deb"
  lifecycle {
    create_before_destroy = true
  }
}
```

The idea is that if you have a typo in the tag, it'll be an error before trying to make changes rather than later. It's a slight abuse of Terraform semantics; in the forthcoming 0.7 there will be a "data" concept (parallel to "resource") that makes this work a little better.